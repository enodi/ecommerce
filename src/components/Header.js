import React from "react";
import { Link } from "react-router-dom";

const Header = ({ handleShowSidebar, cartCount }) => (
  <header>
    <ul>
      <li>
        <Link to="/">Lumin</Link>
      </li>
      <li>
        <Link to="/">Shop</Link>
      </li>
      <li>
        <Link to="/">Help</Link>
      </li>
      <li>
        <Link to="/">Blog</Link>
      </li>
    </ul>
    <ul>
      <li>
        <Link to="/">Account</Link>
      </li>
      <li onClick={handleShowSidebar}>{`Cart ${cartCount}`}</li>
    </ul>
  </header>
);

export default Header;
