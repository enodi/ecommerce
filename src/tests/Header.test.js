import React from "react";
import { shallow } from "enzyme";

import Header from "../components/Header";

let handleShowSidebar, wrapper;
beforeEach(() => {
  handleShowSidebar = jest.fn();
  wrapper = shallow(
    <Header handleShowSidebar={handleShowSidebar} cartCount={4} />
  );
});

test("should render Header component correctly", () => {
  expect(wrapper).toMatchSnapshot();
});

test("should find last text rendered on header", () => {
  expect(wrapper.find("ul").first().find("li").last().text()).toEqual("Blog");
});
