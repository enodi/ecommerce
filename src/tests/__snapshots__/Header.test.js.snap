// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header component correctly 1`] = `
<header>
  <div
    className="header-container"
  >
    <ForwardRef(SvgHamburger)
      className="hamburger"
      style={
        Object {
          "height": 23,
          "width": 23,
        }
      }
    />
    <ul
      className="logo-section"
    >
      <li
        className="logo-container"
      >
        <Link
          to="/"
        >
          <img
            alt="lumin logo"
            className="logo"
            src="logo.png"
          />
        </Link>
      </li>
      <li>
        <Link
          to="/"
        >
          Shop
        </Link>
      </li>
      <li>
        <Link
          to="/"
        >
          Help
        </Link>
      </li>
      <li>
        <Link
          to="/"
        >
          Blog
        </Link>
      </li>
    </ul>
    <ul>
      <li>
        <Link
          to="/"
        >
          Account
        </Link>
      </li>
      <li
        className="cart"
        onClick={[MockFunction]}
      >
        <ForwardRef(SvgShoppingCart)
          style={
            Object {
              "height": 23,
              "width": 23,
            }
          }
        />
        <span
          className="cart--count"
        >
          4
        </span>
      </li>
    </ul>
  </div>
</header>
`;
